get_hrc.py

```diff
@@ -2,88 +2,158 @@
+        except Exception as err:
+            last_err = err
+            time.sleep(5)
+    else:
+        print(f'Warning: failed to open URL {url}: {last_err}')
+        sys.exit(0)
+
+    dat = Table(data)
+
+    return dat
+
+
+def calc_hrc_shield(dat):
+    # For GOES earlier than 16 use columns p5, p6, p7
+    # hrc_shield = (6000 * dat['p5'] + 270000 * dat['p6']
+    #               + 100000 * dat['p7']) / 256.
+    # HRC proxy, GOES-16, TBD
+    hrc_shield = (6000 * dat['p5'] + 270000 * dat['p7']
+                  + 100000 * dat['p9']) / 256.
+    return hrc_shield
+
+
+def format_proton_data(dat, descrs=None):
+    """
+    Manipulate the data and return them in a desired format
+    including columns that the old h5 file format wanted.
+    """
+
+    if descrs is None:
+        with tables.open_file('hrc_shield.h5', mode='r',
+                               filters=tables.Filters(complevel=5, complib='zlib')) as h5:
+            descrs = h5.root.data.dtype
+
+    channels = [chan.lower() for chan in np.unique(dat['channel'])]
+
+    tabs = []
+    for channel in channels:
+        ok = dat['channel'] == channel.upper()
+        t = dat[ok]
+        t.rename_column('flux', channel)
+        tabs.append(t)
+
+    newdat = np.ndarray(len(tabs[0]), dtype=descrs)
+
+    newdat['satellite'] = tabs[0]['satellite']
+
+    time_ref = tabs[0]['time_tag']
+
+    for t in tabs:
+        if not all(t['time_tag'] == time_ref):
+            print(f'Warning: mismatch in channel time column')
+            sys.exit(0)
+        else:
+            for col in t.colnames:
+                if col in channels:
+                    newdat[col] = t[col]
+
+    # Add p11 column and the time columns the old file format wanted
+    times = Time(time_ref)
+    newdat['time'] = times.cxcsec
+    newdat['mjd'] = times.mjd.astype(int)
+    newdat['secs'] = np.array(np.round((times.mjd - newdat['mjd']) * 86400,
+                                        decimals=0)).astype(int)
+    newdat['year'] = [t.year for t in times.datetime]
+    newdat['month'] = [t.month for t in times.datetime]
+    newdat['dom'] = [t.day for t in times.datetime]
+    newdat['hhmm'] = np.array([f"{t.hour}{t.minute:02}" for t in times.datetime]).astype(int)
+    newdat['p11'] = np.full(len(times), -1.0e5)
+
+    hrc_shield = calc_hrc_shield(newdat)
+
+    newdat['hrc_shield'] = hrc_shield
+
+    hrc_bad = (newdat['p5'] < 0) | (newdat['p7'] < 0) | (newdat['p9'] < 0)
+    newdat['hrc_shield'][hrc_bad] = -1.0e5  # flag bad inputs
+
+    return newdat, descrs, hrc_bad
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Archive GOES data and '
+                                     'HRC shield rate proxy')
+    parser.add_argument('--data-dir', type=str,
+                        default='.',
+                        help='Directory for output data files')
+    args = parser.parse_args()
+
+    dat = get_json_data()
+    newdat, descrs, hrc_bad = format_proton_data(dat)
```

Comment:
It looks like format_proton_data uses the hrc_shield.h5 in the current working directory, which is not guaranteed to be a directory that contains hrc_shield.h5 .  There is a chdir later, but the chdir is too late and isn't a great idea anyway.